tests/test_rigid_physics.py

```diff
@@ -2177,18 +2177,37 @@ def test_gravity(show_viewer, tol):
     )
 
     sphere = scene.add_entity(gs.morphs.Sphere())
-    scene.build(n_envs=2)
+    scene.build(n_envs=7)
 
-    scene.sim.set_gravity(torch.tensor([0.0, 0.0, -9.8]), envs_idx=0)
-    scene.sim.set_gravity(torch.tensor([0.0, 0.0, 9.8]), envs_idx=1)
+    scene.sim.set_gravity(torch.tensor([-1.0, 0.0, 0.0]))
+    scene.sim.set_gravity(torch.tensor([[-2.0, 0.0, 0.0]] * 10))
+    scene.sim.set_gravity(torch.tensor([0.0, 0.0, 0.0]), envs_idx=[0, 1])
+    scene.sim.set_gravity(torch.tensor([0.0, 0.0, 100.0]), envs_idx=3)
+    scene.sim.set_gravity(torch.tensor([[0.0, 0.0, -10.0], [0.0, 0.0, -1.0]]), envs_idx=[2, 4])
 
-    for _ in range(200):
-        scene.step()
+    with np.testing.assert_raises(AssertionError):
+        scene.sim.set_gravity(torch.tensor([0.0, -10.0]))
 
-    first_pos = sphere.get_dofs_position()[0, 2]
-    second_pos = sphere.get_dofs_position()[1, 2]
+    with np.testing.assert_raises(AssertionError):
+        scene.sim.set_gravity(torch.tensor([[0.0, 0.0, -10.0], [0.0, 0.0, -10.0]]), envs_idx=1)
+
+    scene.step()
 
-    assert_allclose(first_pos * -1, second_pos, tol=tol)
+    assert_allclose(
+        np.array(
+            [
+                [0.0, 0.0, 0.0],
+                [0.0, 0.0, 0.0],
+                [0.0, 0.0, -10.0],
+                [0.0, 0.0, 100.0],
+                [0.0, 0.0, -1.0],
+                [-2.0, 0.0, 0.0],
+                [-2.0, 0.0, 0.0],
+            ]
+        ),
+        sphere.get_links_acc().squeeze(dim=1),
```

Comment:
Are you sure squeezing is necessary?

Reply:
What about `sphere.get_links_acc().squeeze()`? I find it a bit more intuitive but as you want.